### PR TITLE
Aave Optimism Rewards

### PR DESCRIPTION
**Pull request by iri031**
Aave rewards for Optimism.
- [x] Deployment
- [x] Add to SDK
- [x] Docs

### CODE REVIEW

**Comment by iri031** [comment]
### Deployed on Optimism

**Name**: `AAVE-V3-CLAIM-A`
**Addr**: [0x09cadCBe6Fe623F01B7bC37635b97ABF385d46Dd](https://optimistic.etherscan.io/address/0x09cadCBe6Fe623F01B7bC37635b97ABF385d46Dd#code)
**txn** [here](https://optimistic.etherscan.io/tx/0xe4d9cd9ba932e1a3a742601d0c101a774245d3864793fea3402b533b658c233c)

File: contracts/optimism/connectors/aave/aave-rewards/v3/main.sol [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		address[] memory _rewards = new address[](assets.length);
+
+		(_rewards, _amt) = incentives.claimAllRewards(assets, address(this));
+
+		TokenInterface weth = TokenInterface(wethAddr);
+		uint256 wethAmount = weth.balanceOf(address(this));
+		convertWethToEth(wethAmount > 0, weth, wethAmount);
+
+		_eventName = "LogAllClaimed(address[],uint256[])";
+		_eventParam = abi.encode(assets, _amt);
```
Comment (thrilok209) [inline contracts/optimism/connectors/aave/aave-rewards/v3/main.sol]:
```suggestion
		_eventParam = abi.encode(assets, _rewards, _amts);
```

Comment (iri031) [reply contracts/optimism/connectors/aave/aave-rewards/v3/main.sol]: Done

File: contracts/optimism/connectors/aave/aave-rewards/v3/events.sol [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+contract Events {
+	event LogClaimed(
+		address[] assets,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	);
+
+	event LogAllClaimed(address[] assets, address[] rewards, uint256[] amt);
```
Comment (thrilok209) [inline contracts/optimism/connectors/aave/aave-rewards/v3/events.sol]:
```suggestion
	event LogAllClaimed(address[] assets, address[] rewards, uint256[] amts);
```

File: contracts/optimism/connectors/aave/aave-rewards/v3/main.sol [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	{
+		uint256 _amt = getUint(getId, amt);
+
+		require(assets.length > 0, "invalid-assets");
+
+		_amt = incentives.claimRewards(assets, _amt, address(this), reward);
+
+		TokenInterface weth = TokenInterface(wethAddr);
+		uint256 wethAmount = weth.balanceOf(address(this));
+		convertWethToEth(wethAmount > 0, weth, wethAmount);
```
Comment (thrilok209) [inline contracts/optimism/connectors/aave/aave-rewards/v3/main.sol:43]: @Richa-iitr We only need to do this when `reward` is weth, right? 

@pradyuman-verma ?

Comment (pradyuman-verma) [reply contracts/optimism/connectors/aave/aave-rewards/v3/main.sol:43]: Yes

**Review by pradyuman-verma — APPROVED** [review]

**Review by pradyuman-verma — APPROVED** [review]

**Pull request merged**